File: Java/프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.Arrays;
class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        for(int i=0;i<sizes.length;i++){
            int temp=0;
            if(sizes[i][0]<sizes[i][1]){
                temp = sizes[i][0];
                sizes[i][0] = sizes[i][1];
                sizes[i][1] = temp;
            }
        }
        
        
        return maxSize(sizes, 0)*maxSize(sizes, 1);
    }
    int maxSize(int[][] sizes, int idx){
        int max=0;
        for(int i =0; i<sizes.length;i++){
            if(max<sizes[i][idx]){
                max=sizes[i][idx];
            }
        }
        return max;
    }
}

File: Kotlin/프로그래머스/1/86491. 최소직사각형/최소직사각형.kt
class Solution {
    fun solution(sizes: Array<IntArray>): Int {
        var answer: Int = 0
        var maxWidth = 0
        var maxHight = 0
        //1번 제일 큰 값을 찾아
        //2번 각 배열의 두번째 값들 중 제일 큰 값을 찾아
        sizes.forEach{
            if(it[0] > it[1]) {
                if(it[0] > maxWidth) {
                    maxWidth = it[0]
                }
            } else {
                if(it[1] > maxWidth) {
                    maxWidth = it[1]
                }
            }
        }
        sizes.forEach{
            if(it[0] > it[1]) {
                if(it[1] > maxHight) {
                    maxHight = it[1]
                }
            } else {
                if(it[0] > maxHight) {
                    maxHight = it[0]
                }
            }
        }
        return maxWidth*maxHight
    }
}